packages/ui/src/InteractiveList.tsx

```diff
@@ -0,0 +1,207 @@
+import { IconButton } from './IconButton'
+import { Plus, X } from 'react-feather'
+import styles from './InteractiveList.module.scss'
+import { useUID } from 'react-uid'
+import { FieldArrayRenderProps, FieldValidator, FormikHelpers } from 'formik'
+import { Error, errorId } from './Error'
+import { ExtractKeysOfValueType } from './utils/types'
+
+export type InteractiveListExtraProps = {
+  children?: React.ReactNode | React.ReactNode[]
+  inputControlRef?: Ref<InteractiveListInputRef>
+} & Pick<TextFieldExtraProps<'text'>, 'label' | 'helperText' | 'inputIcon' | 'type' | 'placeholder'>
+
+export type InteractiveListCoreProps<V> = {
+  name: ExtractKeysOfValueType<V, string[]>
+  value: string[]
+  onBlur?: (e: FocusEvent<HTMLTextAreaElement>) => void
+  error?: string
+  validate?: FieldValidator
+  arrayHelpers: FieldArrayRenderProps
+  onError: (str: string) => void
+  validateForm: FormikHelpers<V>['validateForm']
+} & InteractiveListExtraProps
+
+export type InteractiveListProps<V> = InteractiveListExtraProps & InteractiveListCoreProps<V>
+
+export type InteractiveListInputRef = { setValue: (value: string) => void }
+
+export type InteractiveListItemProps = {
+  arrayHelpers: FieldArrayRenderProps
+  content: string
+  error?: string
+  idx: number
+}
+
+export const InteractiveListItem = ({ arrayHelpers, content, error, idx }: InteractiveListItemProps) => {
+  const id = useUID()
+  const errorProps = error
+    ? {
+        'aria-invalid': true,
+        'aria-errormessage': errorId(id),
+      }
+    : {}
+
+  return (
+    <>
+      <div className={styles.content}>
+        <span id={id} {...errorProps}>
+          {content}
+        </span>{' '}
+        <IconButton
+          kind="transparent"
+          ariaLabel="Remove Item"
+          icon={X}
+          size="small"
+          onClick={() => {
+            arrayHelpers.remove(idx)
+          }}
+        />
+      </div>
+      {error && <Error error={error} inputId={id} />}
+    </>
+  )
+}
+
+/**
+ * This component is meant to be used only from InteractiveListFormik
+ * @private
+ */
+const InteractiveList = <V,>({
+  label,
+  inputControlRef,
+  children,
+  error,
+  name,
+  value,
+  helperText,
+  inputIcon,
+  onError,
+  type,
+  validate,
+  arrayHelpers,
+  validateForm,
+}: InteractiveListProps<V>) => {
+  const id = useUID()
+  const [inputValue, setValue] = useState<string>('')
+  // We want to revalidate on change only after un-successful insert
+  const [touched, setTouched] = useState(false)
+
+  const normalizedValue = useMemo(() => inputValue.trim(), [inputValue])
+
+  useImperativeHandle(inputControlRef, () => ({
+    setValue: (value: string) => {
+      setValue(value)
+    },
+  }))
+
+  useEffect(() => {
+    if (touched) {
+      void getValidationError().then((validateError) => {
+        if (validateError) {
+          onError(validateError)
+        }
+      })
+    }
+  }, [touched, inputValue])
+
+  const getValidationError = async () => {
```

Comment:
@jvorcak the only thing that is bothering me just a bit is that now this component is Fromik-bound. What if we lift validation and error handling to `InteractiveListFormik`? Then this component will accept the final error as `error` prop and to trigger the validation chain in parent `InteractiveListFormik` we could expose `onChange`. WDYT?

Reply:
The reason I went this way is that `addValueCallback` - function that adds an actual element to a list uses `arrayHelpers` from Formik -> thus needs to be in a lower component. In this function, I'd like to call validation somehow to make sure the value I add is correct. I could use it as a prop from outer Formik component, but that would make it Formik-bound anyway -> just a bit more complicated.  Or do you have some other idea?

I think if a component is dependent on `arrayHelpers`, it's formik-bound either way.

Reply:
Makes sense. Another option wild be to provide an onValueAdd callback to the parent formik-bound component. Which could run validation and sdd the value if needed. This way the list component could be used with any form library in the future. WDYT?

Reply:
@aigoncharov via ImperativeHandle? Isn't that overusing is bit? However, I can do that if you want.

Reply:
@aigoncharov you can have a look

Reply:
@jvorcak I was about to propose something similar to what you have implemented. Great job!